content/en/docs/developer/e2e-guide.md

```diff
@@ -0,0 +1,70 @@
+---
+draft: false
+linktitle: E2E test guide
+menu:
+  docs:
+    parent: developer guide
+    weight: 15
+title: Run E2E test
+toc: true
+type: docs
+
+---
+End-to-End (E2E) testing is a crucial component in modern software development, designed to simulate user interactions across the entire application to ensure all components and integrations function together seamlessly. By incorporating E2E testing, we can verify that changes in code do not disrupt existing functionalities, maintaining the integrity and reliability of the system even as it evolves. 
+
+## Prerequisites
+
+The Go version in go.mod is 1.22.0
+
+- [Update the go version](https://go.dev/doc/install)
+- Use `go mod tidy`
+
+The following components will be installed while using the shell
+
+- Kind
+- Helm
+- istioctl 
+
+Components that need to be installed by the user
+
+- [Docker](https://docs.docker.com/engine/install/)
```

Comment:
Put the components that user need to install at first.

Also we don't need to provide installation link, as they may change and we don't want to maintain them.